books/kestrel/c/syntax/types.lisp

```diff
@@ -234,7 +236,17 @@
   :short "Check if a type is a standard signed integer type [C17:6.2.5/4]."
   (and (member-eq (type-kind type) '(:schar :sshort :sint :slong :sllong))
        t)
-  :hooks (:fix))
+  :hooks (:fix)
+
+  ///
+
+  (defrule type-standard-signed-integerp-when-type-kind-syntaxp
```

Comment:
I wonder if we should disable these by default, given that the LHS is fairly generic? (They could be put in a rule set, to be enabled for the proofs that need this kind of reasoning.)

Reply:
Are you thinking the rules may be undesirable in some contexts, or that the rules may be expensive?

If the former, I think the rules are universally desirable because they will only apply when the LHS can be rewritten to a ground term (which will be boolean).

If the latter, I assumed the rules wouldn't be too expensive. There is no backchaining, just binding a free variable and then checking the `syntaxp` constraint. Unless the free variable binding is more expensive than I imagine?

Regardless, if you think we should disable these rules by default, I can do so and add them to a ruleset like you suggest.